src/swmain.c

```diff
@@ -157,11 +158,18 @@ static void calculate_lag(void)
 static void new_move(void)
 {
 	int multkey;
+	int multbtn;
 	int tictime;
 
 	/* generate a new move command and save it */
 
 	multkey = Vid_GetGameKeys();
+	if(isGamepadInitted()) {
```

Comment:
Is there any reason you can't just:
```
multkey = Vid_GetGameKeys() | Gamepad_GetGameBtns();
```
?

Then I think you can get rid of all the "LastInputGamepad" stuff entirely.